### DISCUSSION TITLE
Async Enumerables and .net 4.0

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#2788] [Q&A]
I know that not all C#8.0 features will be available on .NET 4.x (for example, default interface methods aren't available).

Will Async Enumerables work on .NET 4.x?

**Comment by huoyaoyuan** [comment]
The [support package](https://www.nuget.org/packages/Microsoft.Bcl.AsyncInterfaces) requires 4.6.1 at least.

**Comment by TonyValenti** [comment]
@huoyaoyuan -
I just installed VS 2019.3 and it seems like you can't use C# 8.0 on a .NET 4.6.1 app.  Am I missing something?

![image](https://user-images.githubusercontent.com/13609558/65516014-d968b980-dea5-11e9-991c-052c35394888.png)

**Comment by huoyaoyuan** [comment]
@TonyValenti At a video in .NET Conf, Mads says that "using new language version (8.0) on older framework" is for pro users, so the selection is removed from IDE. Pro users should read explanations in the link and edit project files manually.

**Comment by YairHalberstadt** [comment]
@tonyValenti

You can manually set the language version in the csproj

**Comment by yaakov-h** [comment]
Note that from my team's testing, if you use C# 8.0 language features in a classic project (rather than the new project system), Intellisense will report (false) errors for every use of an 8.0 language feature.
